java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java

```diff
@@ -60,42 +58,43 @@
   private BytecodeCFGBuilder() {
   }
 
-  public static BytecodeCFG buildCFG(Symbol.MethodSymbol methodSymbol, SquidClassLoader classLoader) {
-    try(InputStream is = classLoader.getResourceAsStream(Convert.bytecodeName(((JavaSymbol.TypeJavaSymbol) methodSymbol.owner()).getFullyQualifiedName()) + ".class")) {
+  public static BytecodeCFG buildCFG(String signature, SquidClassLoader classLoader) {
+    try(InputStream is = classLoader.getResourceAsStream(Convert.bytecodeName(signature.substring(0, signature.indexOf('#'))) + ".class")) {
       byte[] bytes = ByteStreams.toByteArray(is);
       // to read bytecode with ASM not supporting Java 9, we will set major version to Java 8
       if (Java9Support.isJava9Class(bytes)) {
         Java9Support.setJava8MajorVersion(bytes);
       }
-      return buildCFG(methodSymbol, bytes);
+      return buildCFG(signature, bytes);
     } catch (IOException e) {
       throw Throwables.propagate(e);
     }
   }
 
   @VisibleForTesting
-  static BytecodeCFG buildCFG(Symbol.MethodSymbol methodSymbol, byte[] bytes) {
+  static BytecodeCFG buildCFG(String sign, byte[] bytes) {
     ClassReader cr = new ClassReader(bytes);
-    BytecodeCFGMethodVisitor methodVisitor = new BytecodeCFGMethodVisitor(methodSymbol);
+    BytecodeCFGMethodVisitor methodVisitor = new BytecodeCFGMethodVisitor();
     cr.accept(new ClassVisitor(Opcodes.ASM5) {
       @Override
       public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
-        // TODO : name matching is not sufficient in case of overloading.
-        if (name.equals(methodSymbol.name())) {
+        if (name.equals(sign.substring(sign.indexOf('#') + 1, sign.indexOf('('))) && desc.equals(sign.substring(sign.indexOf('(')))) {
+          methodVisitor.isStaticMethod = (access & Flags.STATIC) != 0;
+          methodVisitor.isVarArgs = (access & Flags.VARARGS) != 0;
           return methodVisitor;
         }
         return null;
       }
     }, ClassReader.SKIP_DEBUG | ClassReader.SKIP_FRAMES);
-    return methodVisitor.cfg;
+    return methodVisitor.cfg();
   }
 
   public static class BytecodeCFG {
     public List<Block> blocks;
-    Symbol.MethodSymbol methodSymbol;
+    public boolean isStaticMethod;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make isStaticMethod a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)